shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h

```diff
@@ -375,6 +385,13 @@ typedef void (*FlutterPluginRegistrantCallback)(NSObject<FlutterPluginRegistry>*
  */
 @protocol FlutterAppLifeCycleProvider
 - (void)addApplicationLifeCycleDelegate:(NSObject<FlutterApplicationLifeCycleDelegate>*)delegate;
+
+/**
+ * Implement this in the `UIAppDelegate` of your app to enable Flutter plugins to receive
+ * calls as `UNUserNotificationCenterDelegate` when they are added to
+ * `addApplicationLifeCycleDelegate`.
+ */
+- (void)registerAsUserNotificationCenterDelegate API_AVAILABLE(ios(10));
```

Comment:
I'm a bit confused by this pattern.  Implementing this method in a UIAppDelegate doesn't enable Flutter plugins to receive notification calls: adding `-addApplicationLifeCycleDelegate:` with delegates that implement `-userNotificationCenter:didReceiveNotificationResponse::` PLUS either setting the FlutterAppDelegate to the UNUserNotificationCenter delegate or calling this method is what would make the plugins receive notifications.

And it's not obvious that calling `+[FlutterAppDelegate registerAsUserNotificationCenterDelegate]` will override whatever delegate was already set on UNUserNotificationCenter.

Maybe FlutterAppLifeCycleProvider should implement UNUserNotificationCenterDelegate, registerAsUserNotificationCenterDelegate  be removed, and then the FlutterAppLifeCycleProvider 
 documentation updated to instruct the developer to set the FlutterAppLifeCycleProvider (FlutterAppDelegate, etc) to the UNUserNotificationCenter's delegate to get plugin notification calls.